dask/tests/test_order.py

```diff
@@ -877,3 +877,312 @@ def test_array_store_final_order(tmpdir):
     connected_max = max([v for k, v in o.items() if k in connected_stores])
     disconnected_min = min([v for k, v in o.items() if k in disconnected_stores])
     assert connected_max < disconnected_min
+
+
+def test_eager_to_compute_dependent_to_free_parent():
+    r"""https://github.com/dask/dask/pull/7929
+
+    This graph begins with many motifs like the following:
+
+    |      |
+    c1    c2
+      \ /
+       b
+       |
+       a
+
+    We want to compute c2 and c3 pretty close together, because if we choose to
+    compute c1, then we should also compute c2 so we can release b.  Being
+    greedy here allows us to release memory sooner and be more globally optimal.
+    """
+    dsk = {
+        "join-ac838b48-7eab-4ed5-96b1-42ca515d6e44": (
```

Comment:
This is a very nice test case, but the text is kind of long.  Is it worth compacting?

Reply:
If anyone other than you were to submit this then yes, we would ask them to simplify.  As long as you don't plan to get hit by a bus though I'm fine with whatever you do here :)

Reply:
Gotcha.  I'll try to avoid said busses.  And bears.  And poisonous snakes.  All good! ;)